tests/integration/conftest.py

```diff
@@ -139,14 +138,24 @@ def flow(run_dir, test_dir):
 
 @pytest.fixture(scope='module')
 def mod_scheduler():
-    """Return a scheduler object for a flow."""
-    return _make_scheduler
+    """Return a Scheduler object for a flow.
+
+    Usage: see scheduler() below
+    """
+    with _make_scheduler() as _scheduler:
+        yield _scheduler
```

Comment:
This slightly awkward syntax is how you reuse a "factory as fixture" that yields instead of returns